### PR TITLE
Scripts and documentation to release to npmjs.org

### PR DESCRIPTION
**Pull request by aliuk2012**
This PR will enable us to publish digitalmarketplace-govuk-frontend to npm. I have also tried to document the steps that we will follow to publish a release.

I did try to follow and reuse most of [govuk-frontend's](https://github.com/alphagov/govuk-frontend) process

Ticket :https://trello.com/c/KBZFQOfK

### CODE REVIEW

File: bin/release.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+if [[ $continue_prompt != 'y' ]]; then
+    echo "Cancelling release, if this was a mistake, try again and use 'y' to continue."
+    exit 0
+fi
+
+NPM_USER=$(npm whoami)
+if ! [ "digital-marketplace" == "$NPM_USER" ]; then
+    echo "⚠️   FAILURE: You are not logged in with the correct user."
+    echo "Please login using the details found in https://github.com/alphagov/digitalmarketplace-credential"
```
Comment (risicle) [inline bin/release.sh]: `digitalmarketplace-credentials`

File: bin/release.sh [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    exit 1
+fi
+
+echo "📦  Publishing package..."
+
+# Try publishing
+cd package
+npm publish $NPM_ARGS
+echo "🗒 Package published!"
+cd ..
```
Comment (risicle) [inline bin/release.sh]: It's often nicer to use `pushd` and `popd` for this kinda stuff...

File: docs/publishing.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+5. Update [`CHANGELOG.md`](../CHANGELOG.md) "Unreleased" heading with the new version number.
+
+6. Update [`package/package.json`](../package/package.json) version with the new version number.
+
+7. Save the changes. Do not commit.
+
+8. Run `npm run pre-release`, you will be prompted to continue or cancel.
+
+9. (Optional) Test in with any Digital Marketplace frontend application
```
Comment (jonodrew) [inline docs/publishing.md]: Test it?

File: docs/publishing.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  - add release notes from changelog
+  - add a summary of highlights
+  - publish release
+
+16. Log out from npm
+```bash
+npm logout
+```
+
+19. Add a card in Trello to update each Digital Marketplace frontend application.
```
Comment (jonodrew) [inline docs/publishing.md]: Should this be 17?

File: docs/publishing.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+12. Checkout **master** and pull the latest changes.
+
+13. Log into npm `npm login`, using team [credentials](https://github.com/alphagov/digitalmarketplace-credentials/tree/master/pass/npmjs.org).
+
+14. Run `npm run release`, you will be prompted to continue or cancel.
+
+15. Create a release in the [Github interface](https://github.com/alphagov/digitalmarketplace-govuk-frontend/releases/new)
+  - select the latest tag version
+  - set "Digtial Marketplace GOV.UK Frontend release v[version-number]" as the title
```
Comment (jonodrew) [inline docs/publishing.md]: Tiny spelling error here - Digital rather than Digtial

File: docs/CHANGELOG_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+  ([PR #N](https://github.com/alphagov/govuk-frontend/pull/N))
+
+🔧 Fixes:
+
+- Pull Request Title goes here
+
+  Description goes here (optional)
+
+  ([PR #N](https://github.com/alphagov/govuk-frontend/pull/N))
```
Comment (katstevens) [inline docs/CHANGELOG_TEMPLATE.md]: Liking these ✅

File: bin/release.sh [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+if [ $(git tag -l "$TAG") ]; then
+    echo "⚠️  Tag $TAG already exists"
+    exit 1
+else
+    echo "🗒  Tagging repo using tag version: $TAG ..."
+    git tag $TAG -m "Digital Marketplace frontend release $TAG"
+    git push --tags
+    echo "🗒 Tag $TAG created and pushed to remote."
+    echo "To update run 'npm install dmp-frontend@$TAG'"
```
Comment (katstevens) [inline bin/release.sh]: `dmp-frontend` -> `digitalmarketplace-govuk-frontend`? Or am I getting confused...

File: bin/release.sh (hunk 1 above)
Comment (aliuk2012) [reply bin/release.sh]: Good spot 👍

File: bin/release.sh (hunk 2 above)
Comment (aliuk2012) [reply bin/release.sh]: Thanks I've learnt something new 😎, I haven't come across both commands before but I think in this instance I'll still leave it with `cd`

**Pull request merged**